src/main/kotlin/com/grepp/nbe1_3_team04/vote/service/VoteServiceImpl.kt

```diff
@@ -0,0 +1,276 @@
+    }
+
+    @Transactional
+    override fun deleteVote(voteId: Long, member: Member): Long {
+        val vote: Vote = getVoteByVoteId(voteId)
+        vote.checkWriterFromMemberId(getMemberIdFrom(member))
+        cancelTaskInSchedulerFromVoteId(voteId)
+        voteRepository.delete(vote)
+        return voteId
+    }
+
+    @Transactional
+    override fun createChoice(request: ChoiceCreateServiceRequest, voteId: Long, member: Member): VoteResponse {
+        val memberId: Long = getMemberIdFrom(member)
+        val vote: Vote = getVoteByVoteId(voteId)
+
+        val choices: List<Choice> = createChoice(request, memberId)
+        choiceRepository.saveAll(choices)
+
+        val voteItemResponses: List<VoteItemResponse> = convertVoteItemsToResponseFrom(vote.voteItems)
+        return VoteResponse.of(vote, voteItemResponses)
+    }
+
+    @Transactional
+    override fun deleteChoice(voteId: Long, member: Member): VoteResponse {
+        val memberId: Long = getMemberIdFrom(member)
+        val vote: Vote = getVoteByVoteId(voteId)
+
+        val choices: List<Choice> = choiceRepository.findByMemberIdAndVoteId(memberId, voteId)
+
+        choiceRepository.deleteAllInBatch(choices)
+        val voteItemResponses: List<VoteItemResponse> = convertVoteItemsToResponseFrom(vote.voteItems)
+        return VoteResponse.of(vote, voteItemResponses)
+    }
+
+    @Transactional
+    override fun updateVote(serviceRequest: VoteUpdateServiceRequest, voteId: Long, member: Member): VoteResponse {
+        val memberId: Long = getMemberIdFrom(member)
+        val vote: Vote = getVoteByVoteId(voteId)
+
+        vote.update(serviceRequest.title, serviceRequest.endAt, memberId)
+        cancelTaskInSchedulerFromVoteId(voteId)
+        addVoteTaskToTaskSchedule(vote)
+
+        val voteItemResponses: List<VoteItemResponse> = convertVoteItemsToResponseFrom(vote.voteItems)
+        return VoteResponse.of(vote, voteItemResponses)
+    }
+
+    @Transactional
+    override fun closeVote(voteId: Long, member: Member): VoteResponse {
+        val vote: Vote = getVoteByVoteId(voteId)
+        vote.checkWriterFromMemberId(getMemberIdFrom(member))
+        if (vote.voteItems[0] is VoteItemDate) {
+            makeReservation(vote)
+        }
+        vote.updateVoteStatusToClose()
+        cancelTaskInSchedulerFromVoteId(voteId)
+        return VoteResponse.of(vote, convertVoteItemsToResponseFrom(vote.voteItems))
+    }
+
+    override fun getAllVotes(teamId: Long): List<AllVoteResponse> {
+        validateTeamByTeamId(teamId)
+        val votes: List<Vote> = voteRepository.findAllByTeamId(teamId)
+        return votes.map(AllVoteResponse::from)
+            .toList()
+    }
+
+    private fun createVoteItemDate(request: VoteDateCreateServiceRequest, savedVote: Vote): List<VoteItemDate> {
+        return voteItemRepository.saveAll(request.choices.stream()
+            .map { choice -> VoteItemDate.create(savedVote, choice) }
+            .toList())
+    }
```

Comment:
아까 말씀하신게 이런 부분들이군요!
Java Stream Api 대신 Kotlin에서 map()을 직접 사용하는 것을 원한다면 다음과 같이 작성할 수 있을 것 같습니다.
```suggestion
    private fun createVoteItemDate(request: VoteDateCreateServiceRequest, savedVote: Vote): 
    List<VoteItemDate> {
        return voteItemRepository.saveAll(
            request.choices.map { choice -> VoteItemDate.create(savedVote, choice) }
        )
    }
```
하지만 문제가 생길 수 있으니... 일단 그대로 merge하고 나중에 리펙토링 하는 것을 추천할게요!

Reply:
이미 민우님께서 작성하셨네요! ㅎㅎ

Reply:
감사하게도 알려주셔서 바꿨습니다 !

한번에 보고 알아채시네요 코틀린 고수가 되셨네요.. 일단 알려주신대로 리팩토링 해두었습니다 !

Reply:
Java Stream Api를 잘 못쓰다보니.. 이런 점은 오히려 Kotlin이 편하다고 생각이 드네요 ㅎㅎ